app/src/main/java/swati4star/createpdf/fragment/HomeFragment.java

```diff
@@ -544,11 +549,61 @@ public void onItemClick(int position) {
             case 1:
                 cropImages();
                 break;
+            case 2:
+                    compressImage();
+                    break;
             default:
                 break;
         }
     }
 
+    private void compressImage()  {
+
+        if (mTempUris.size() == 0) {
+            Snackbar.make(Objects.requireNonNull(mActivity).findViewById(android.R.id.content),
+                    R.string.snackbar_no_images,
+                    Snackbar.LENGTH_LONG).show();
+            return;
+        }
+
+        final MaterialDialog dialog = new MaterialDialog.Builder(mActivity)
+                .title(R.string.compress_image)
+                .customView(R.layout.compress_image_dialog, true)
+                .positiveText(android.R.string.ok)
+                .negativeText(android.R.string.cancel)
+                .build();
+
+        mPositiveAction = dialog.getActionButton(DialogAction.POSITIVE);
+        mPasswordInput = dialog.getCustomView().findViewById(R.id.quality);
+        mPasswordInput.setText(mQuality);
+        mPasswordInput.addTextChangedListener(
+                new TextWatcher() {
+                    @Override
+                    public void beforeTextChanged(CharSequence s, int start, int count, int after) {
+                    }
+
+                    @Override
+                    public void onTextChanged(CharSequence s, int start, int before, int count) {
+                        mPositiveAction.setEnabled(s.toString().trim().length() > 0);
+                    }
+
+                    @Override
+                    public void afterTextChanged(Editable input) {
+                            mQuality = input.toString();
+                            Log.e("log","Imput : "+mQuality);
```

Comment:
Change tag form log.e to log.v

Reply:
Change quality from string to int here only. & Show the error if input is not a valid integer or outside the range.